src/Components/Patient/PatientRegister.tsx

```diff
@@ -937,6 +962,17 @@ export const PatientRegister = (props: PatientRegisterProps) => {
   const renderMedicalHistory = (id: number, title: string, field: any) => {
     const checkboxField = `medical_history_check_${id}`;
     const textField = `medical_history_${id}`;
+    const filteredCancerHistoryChoices = CANCER_HISTORY_CHOICES.filter((i) => {
+      const gender = field("gender").value;
+      if (gender === 1) {
+        return i.id !== 1 && i.id !== 5;
+      } else if (gender === 2) {
+        return i.id !== 8;
+      } else {
+        return true;
+      }
```

Comment:
code is not readable. Consider adding comments to define what these numbers represent.